docs/developing.rst

```diff
@@ -742,3 +742,246 @@ Note the call to :func:`toil.job.Job.encapsulate` creates the
+step can be replaced with ``python setup.py install``. Note that ``python
+setup.py develop`` would not work here because it does not copy the source code
+but creates .egg-links instead, which Toil is not able to hot-deploy.
+
+.. _setuptools: http://setuptools.readthedocs.io/en/latest/index.html
+
+The main caveat to this solution is that the external dependencies may not
+contain native code, they must be pure Python. If you have dependencies that
+rely on native code, you must manually install them on each worker.
+
+
+Relying on shared filesystems
+-----------------------------
+
+Bare-metal clusters typically mount a shared file system like NFS on each node.
+If every node has that file system mounted at the same path, you can place your
+project on that shared filesystem and run your user script from there.
+Additionally, you can clone the Toil source tree into a directory on that
+shared file system and you won't even need to install Toil on every worker. Be
+sure to add both your project directory and the Toil clone to ``PYTHONPATH``. Toil
+replicates ``PYTHONPATH`` from the leader to every worker.
+
+TODO: Does a virtualenv on a shared filesystem work?
+
+
+Deploying Toil
+==============
+
+We've looked at various ways of installing your workflow on the leader such
+that Toil can replicate it to the workers and load the job definitions there.
+But what about Toil itself? Unless you are running your workflow in single
+machine mode (the default) or on a cluster where every node mounts a shared
+file system at the same path, Toil somehow needs to be made available on each
+worker. Unfortunately, hot-deployment only works for the user script/module and
+its dependencies, not for Toil itself. Generally speaking, you or your admin
+will need to manually :ref:`install <installation>` Toil on every cluster node
+you intend to run Toil jobs on.
+
+The Toil team is eagerly working to ameliorate this. Toil 3.5.0 will contain
+the Toil Appliance, a Docker image that contains Mesos and Toil. You can use
+this image to run the Toil Appliance locally without the need to install
+anything. Only Docker is required. Inside the appliance you can then run a
+workflow in single machine mode. From the appliance, you will also be able to
+provision clusters of VMs in the cloud. Initially this will support Amazon EC2
+only, but Google Cloud and Microsoft Azure will soon follow.
+
+For the current stable release (3.3.x), you can use `CGCloud`_ to provision a
+cluster of Amazon EC2 instances with Toil and Mesos on them. The ``contrib``
+directory of the Toil contains a Adam Novak's Azure resource template with
+which you can deploy a Toil cluster in Azure. With CGCloud you would typically
+provision a static cluster of either spot or on-demand instances, or a mix.
+This is explained in more detail in section :ref:`installation`.
+
+.. _CGCloud: https://github.com/BD2KGenomics/cgcloud
+
+
+Depending on Toil
```

Comment:
Call this section out earlier, since it is:

1. Important
2. Non-obvious, and
3. At the end of a long section of the docs